api/server/middleware_auth.go

```diff
@@ -206,6 +215,14 @@ func (a *authMiddleware) deleteWithAuth(w http.ResponseWriter, r *http.Request,
 		json.NewEncoder(w).Encode(volumeResponse)
 		return
 	}
+	if secretName == "" || secretContext == "" {
+		errorMessage := "Internal error, no secret found in the annotations of the persistent volume claim"
```

Comment:
@lpabon If a cluster goes from no auth -> auth, then all the Kubernetes requests are going to get this error. And we will not be able to delete a volume through Kubernetes. Is it ok to just doc this edge case?

Reply:
offline discussion we decided that it is ok to fail the PVC delete request coming from Kubernetes, but provide the SDK as the method of cleaning up the volumes in this situation.